apps/prairielearn/src/pages/instructorAssessmentGroups/instructorAssessmentGroups.html.ts

```diff
@@ -0,0 +1,642 @@
+    <form name="delete-member-form" method="POST">
+      <div class="form-group">
+        <label for="delete-member-form-${iRow}">UID:</label>
+        <select class="form-control" name="user_id" id="delete-member-form-${iRow}">
+          ${row.users.map((user) => {
+            return html` <option value="${user.user_id}">${user.uid}</option> `;
+          })}
+        </select>
+      </div>
+      <input type="hidden" name="__action" value="delete_member" />
+      <input type="hidden" name="__csrf_token" value="${resLocals.__csrf_token}" />
+      <input type="hidden" name="group_id" value="${row.group_id}" />
+      <button
+        type="button"
+        class="btn btn-secondary"
+        onclick="$('#row${iRow}Popoverdeletemember').popover('hide')"
+      >
+        Cancel
+      </button>
+      <button type="submit" class="btn btn-danger" ${row.users.length > 0 ? '' : 'disabled'}>
+        Delete
+      </button>
+    </form>
+  `;
+}
+
+function uploadAssessmentGroupsModal(csrfToken: string) {
+  return html`
+    <div class="container-fluid">
+      <div
+        class="modal fade"
+        id="uploadAssessmentGroupsModal"
+        tabindex="-1"
+        role="dialog"
+        aria-labelledby="uploadAssessmentGroupsModalLabel"
+      >
+        <div class="modal-dialog" role="document">
+          <div class="modal-content">
+            <div class="modal-header">
+              <h1 class="h4 modal-title" id="uploadAssessmentGroupsModalLabel">
+                Upload new group assignments
+              </h1>
+            </div>
+            <div class="modal-body">
+              <p>Upload a CSV file in the format of:</p>
+            </div>
+            <div class="modal-body">
+              <code class="mb-0 text-dark">
+                groupName,UID<br />
+                groupA,one@example.com<br />
+                groupA,two@example.com<br />
+                groupB,three@example.com<br />
+                groupB,four@example.com</code
+              >
+              <!-- Closing code tag not on its own line to improve copy/paste formatting -->
+            </div>
+            <div class="modal-body">
+              <p>
+                The <code>groupname</code> column should be a unique identifier for each group. To
+                change the grouping, link uids to the groupname.
+              </p>
+              <form name="upload-assessment-group-form" method="POST" enctype="multipart/form-data">
+                <div class="form-group">
+                  <div class="custom-file">
+                    <input
+                      type="file"
+                      accept=".csv"
+                      name="file"
+                      class="custom-file-input"
+                      id="uploadAssessmentGroupsFileInput"
+                    />
+                    <label class="custom-file-label" for="uploadAssessmentGroupsFileInput"
+                      >Choose CSV file</label
+                    >
```

Comment:
Nit (similarly elsewhere):

```suggestion
                    <label class="custom-file-label" for="uploadAssessmentGroupsFileInput">
                      Choose CSV file
                    </label>
```